### PR TITLE
webadmin: add dataCenterId field to SD entity object

### PR DESCRIPTION
**Pull request by mkemel**
Adding dataCenterId field to the Storage Domain entity, in order to
fetch Storage Domain object from datacenter/id/storagedomains URL
with single request, while building Storage Domain Connections dialog

Bug-Url: https://bugzilla.redhat.com/977379

### CODE REVIEW

File: frontend/webadmin/modules/webadmin/src/main/java/org/ovirt/engine/ui/webadmin/plugin/entity/EntityObject.java [hunk 1]
```diff
@@ -65,6 +65,7 @@ protected EntityObject() {
         } else if (businessEntity instanceof StorageDomain) {
             obj.setValueAsString("name", ((StorageDomain) businessEntity).getStorageName()); //$NON-NLS-1$
             obj.setValueAsString("type", ((StorageDomain) businessEntity).getStorageType().toString().toLowerCase()); //$NON-NLS-1$
+            obj.setValueAsString("dataCenterId", ((StorageDomain) businessEntity).getStoragePoolId().toString()); //$NON-NLS-1$
```
Comment (rszwajko) [inline frontend/webadmin/modules/webadmin/src/main/java/org/ovirt/engine/ui/webadmin/plugin/entity/EntityObject.java:68]: it's not clear to me if the ID will always exist. If we have doubts then it's worth to add null check.

Comment (mkemel) [reply frontend/webadmin/modules/webadmin/src/main/java/org/ovirt/engine/ui/webadmin/plugin/entity/EntityObject.java:68]: Seems like across the code it is accessed without null check...

**Review by barpavel — APPROVED** [review]

**Review by ljelinkova — APPROVED** [review]

File: frontend/webadmin/modules/webadmin/src/main/java/org/ovirt/engine/ui/webadmin/plugin/entity/EntityObject.java (hunk 1 above)
Comment (ahadas) [reply frontend/webadmin/modules/webadmin/src/main/java/org/ovirt/engine/ui/webadmin/plugin/entity/EntityObject.java:68]: @mkemel but a storage domain may not be attached to any data center, not in the use-case you wrote in the PR description but generally that can happen

Comment (mkemel) [reply frontend/webadmin/modules/webadmin/src/main/java/org/ovirt/engine/ui/webadmin/plugin/entity/EntityObject.java:68]: Hmm, so maybe something like that?
```suggestion
            obj.setValueAsString("dataCenterId", String.valueOf(((StorageDomain) businessEntity).getStoragePoolId())); //$NON-NLS-1$
```

**Review by rszwajko — APPROVED** [review]

**Review by ahadas — APPROVED** [review]

**Pull request merged**